core/src/com/unciv/ui/screens/worldscreen/unit/actions/UnitActionModifiers.kt

```diff
@@ -18,16 +21,69 @@ object UnitActionModifiers {
             .filter { unique -> unique.conditionals.none { it.type == UniqueType.UnitActionExtraLimitedTimes } }
             .filter { canUse(unit, it) }
 
-    private fun getMovementPointsToUse(actionUnique: Unique): Int {
+    private fun getMovementPointsToUse(unit: MapUnit, actionUnique: Unique, defaultAllMovement: Boolean = false): Int {
+        if (actionUnique.conditionals.any { it.type == UniqueType.UnitActionMovementCostAll })
+            return unit.getMaxMovement()
         val movementCost = actionUnique.conditionals
             .filter { it.type == UniqueType.UnitActionMovementCost }
             .minOfOrNull { it.params[0].toInt() }
-        if (movementCost != null) return movementCost
-        return 1
+        val movementCostRequired = actionUnique.conditionals
+            .filter { it.type == UniqueType.UnitActionMovementCostRequired }
+            .minOfOrNull { it.params[0].toInt() }
+        if (movementCost != null && movementCostRequired != null) {
```

Comment:
Sounds to me like sequenceOf(a,b).filterNotNull.maxOrNull

Reply:
Reworked